Add full models to regression suite

rpavlovicTT opened the pull request:
### Ticket
Closes https://github.com/tenstorrent/tt-mlir/issues/4121

### What's changed
Added full models to test suite. They are written in TTIR form produced by ForgeFE. It's not ideal as ForgeFE may change its output at some point, but we can update tests on a schedule. It's more important to have regression tests in place right now, to avoid pain when debugging regressions caught only in ForgeFE after uplifting. 


### Checklist
- [x] New/Existing tests provide coverage for changes

svuckovicTT commented:
I think having these tests is going to make things easier to test, but there's a maintenance cost to it, so sharing a couple considerations:
1. How do we catch situations where FFE changes the IR? Schedule kiiind of works. How often do we update them?
2. Who does the update i.e. who owns these tests?
3. Are there instructions on how to generate these tests? What if the tests get relocated?

Maybe some of these issues could be solved by having a CI job in FFE as well?

sdjordjevicTT reviewed (changes requested):
I’d like to discuss this approach before we move forward with merging. There are a few concerns:
	•	Even small changes in ttir or ttcore require widespread updates across the large, text-based models.
	•	These models can easily fall out of sync with the actual Forge-FE models.
	•	It’s unclear how long these tests take to run—could they significantly impact CI time?

Overall, this feels like a workaround driven by the structure of our repo setup, where we maintain separate modules in separate repos. Because of that, I don’t think this is the right place for these tests.

odjuricicTT commented on .github/workflows/build-and-test.yml:
```diff
@@ -376,6 +376,30 @@ jobs:
 
+    # Regression tests for models - run llvm-lit with regression flag and execute flatbuffer
+    # To run locally: ./test/ttmlir/Silicon/TTNN/n150/optimizer/regression_tests/run_regression_tests.sh
+    - name: Run Regression Tests
```
Should the name be more specific? Can we call this "Optimizer Model Tests" or "Benchmark Tests"

rpavlovicTT replied:
Renamed

rpavlovicTT commented:
@odjuricicTT 

Pushed following changes:
- Reduced tests to: resnet, yolo_v8 and segformer.
- Added CMake build target `check-optimizer-model-regression` for cleaner
  test execution.
- Renamed workflow and references from "regression tests" to "optimizer models regression tests" for clarity.
- Updated CI workflow to use CMake target instead of duplicating script logic.

rpavlovicTT commented on test/lit.cfg.py:
```diff
@@ -43,14 +43,18 @@
     config.available_features.add("opmodel")
 
+# Regression tests are optionally enabled for specific CI jobs via environment variable.
+if os.getenv("TTMLIR_ENABLE_REGRESSION_TESTS", "").upper() in ["1", "ON", "TRUE"]:
```
Added build target which is triggered from CI and can be run locally. Should be easy to reproduce.
